src/soep_preparation/utilities/general.py

```diff
@@ -0,0 +1,57 @@
+"""General utility functions."""
+
+from importlib.util import module_from_spec, spec_from_file_location
+from pathlib import Path
+from types import ModuleType
+
+
+def get_script_names(directory: Path) -> list[str]:
+    """Get the names of all scripts in the given directory.
+
+    Args:
+        directory: The directory containing scripts.
+
+    Returns:
+        A list of script names.
+    """
+    return [
+        file.stem
+        for file in directory.glob("*.py")
+        if file.name not in ["__init__.py", "task.py"]
+    ]
+
+
+def get_file_names(directory: Path) -> list[str]:
```

Comment:
Now I begin to get an understanding of what was meant above. As general feedback, please avoid general terms ("files"), which acquire special meaning because of the context.

```suggestion
def get_stems_if_corresponding_dataset_exists(directory: Path) -> list[str]:
```

or so.

Reply:
Correct, my bad. `datafile` then. `file` is simply too generic, only becomes clear when you have familiarised yourself with the entire context.